applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
     }
 
     /**
-     * Sets key value. Label value for the configurationSetting is optional, if not specified or label=%00 it implies null label.
-     * @param configurationSetting key and value to set
-     * @return ConfigurationSetting that was created or updated
+     * Gets the default pipeline policies.
+     * TODO (conniey): This is going to change when we move to a builder pattern.
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting) {
-        return setKeyValue(configurationSetting, null);
+    public static List<HttpPipelinePolicy> getDefaultPolicies(String connectionString) {
+        final ApplicationConfigCredentials credentials = new ApplicationConfigCredentials(connectionString);
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
+        policies.add(new RequestIdPolicy());
+        policies.add(new RetryPolicy());
+        policies.add(new ConfigurationCredentialsPolicy(credentials));
+
+        return policies;
     }
 
     /**
-     * Sets key value. Label value for the configurationSetting is optional, if not specified or label=%00 it implies null label.
-     * @param configurationSetting key and value to set
+     * Adds a configuration value in the service if that key and label does not exist.
+     *
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     * </p>
+     *
+     * @param configurationSetting The key, value, and label to set.
      * @return ConfigurationSetting that was created or updated
+     * @throws com.azure.common.http.rest.RestException when a ConfigurationSetting with the same key and label exists.
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> addKeyValue(ConfigurationSetting configurationSetting) {
         Validator.validate(configurationSetting);
-        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters().withValue(configurationSetting.value())
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
```

Comment:
`parameters` sounds weird here, knowing that is actually just properties of a ConfigurationSetting.
Is there a way that we could just pass the object that the user is already sending?

Reply:
I see... it is because the `setKey signature` ....

Reply:
I'm not sure about this because if the user populated the etag, last_modified, locked, what would the expected behaviour be if we passed in the entire object?

Reply:
Those 3 values will be ignored by the server. It is actually better if the user can't set them so there isn't confusion thinking that they are going to get overwritten or something